ros2snap

```diff
@@ -1,112 +1,345 @@
-	snappy build install_isolated
-fi
-
-
-echo "Successfully built $1 snap"
+# If building from source (default), run at the root of a catkin workspace
+# TODO Unset env vars
+
+import bloom.generators.common
+import catkin_pkg.packages
+
+import apt
+import argparse
+import os
+import shutil
+import stat
+import sys
+
+core_rosdeps = ['rosclean', 'rosmaster', 'rosout']
+
+def check_create_dir(dirname):
+  if not os.path.exists(dirname):
+    os.makedirs(dirname)
+
+# Resolve and recursively copy dependencies
+class SnappyBuilder:
+  def __init__(self, package_key, packages, pkg_root):
+    self.copied_packages = set()
+    self.package_key = package_key
+    self.package_key_final = package_key.replace("_", "-").lower()
+    self.package = packages[package_key]
+    if self.package is None:
+      print "Requested package " + self.package_key + " not found, abort."
+      sys.exit(-1)
+
+    self.distro = os.getenv("ROS_DISTRO", "jade")
+
+    self.snappy_root = "snappy_build/" + self.package_key_final + "/"
+    self.snappy_meta_dir = self.snappy_root + "meta/"
+    self.snappy_bin_dir = self.snappy_root + "bin/"
+    self.pkg_root = pkg_root
+    print "Package root: " + pkg_root
+
+    setup = ""
+
+
+    # workaround for folder in setup.bash not registering
+    ros_setup_dir = "$mydir/opt/ros/%s" % self.distro
+    setup += "export _CATKIN_SETUP_DIR=%s\n" % ros_setup_dir
+    setup += ". %s/setup.sh\n" % ros_setup_dir
+    # Can't seem to overlay install/setup.sh and opt/ros/indigo/setup.sh
+
+    if self.pkg_root == "install/":
+      #local_setup_dir = "$mydir/%s" % self.pkg_root
+      setup += """export LD_LIBRARY_PATH=$mydir/install/lib:$mydir/install/lib/x86_64-linux-gnu:$LD_LIBRARY_PATH
+export ROS_PACKAGE_PATH=$mydir/install/share:$mydir/install/stacks:$ROS_PACKAGE_PATH
+export PATH=$mydir/install/bin:$PATH
+export PYTHONPATH=$mydir/install/lib/python2.7/dist-packages:$PYTHONPATH
+export PKG_CONFIG_PATH=$mydir/install/lib/pkgconfig:$mydir/install/lib/x86_64-linux-gnu/pkgconfig:PKG_CONFIG_PATH
+export CMAKE_PREFIX_PATH=$mydir/install:$CMAKE_PREFIX_PATH
+"""
+      #setup += "export _CATKIN_SETUP_DIR=%s\n" % local_setup_dir[:len(local_setup_dir)-1]
+      #setup += ". %ssetup.sh\n" % local_setup_dir
+
+    self.environment_script = """#!/bin/bash
+mydir=$(dirname $(dirname $(builtin cd "`dirname "${BASH_SOURCE[0]}"`" > /dev/null && pwd)))
+%s
+export ROS_MASTER_URI=http://localhost:11311
+export LD_LIBRARY_PATH=$mydir/usr/lib/x86_64-linux-gnu:$mydir/usr/lib:$LD_LIBRARY_PATH
+export PATH=$mydir/usr/bin:$PATH
+export PYTHONPATH=$mydir/usr/lib/python2.7/dist-packages:$PYTHONPATH
+export PKG_CONFIG_PATH=$mydir/usr/lib/pkgconfig:$mydir/usr/lib/x86_64-linux-gnu/pkgconfig:$PKG_CONFIG_PATH
+""" % setup
+
+    self.cache = apt.Cache()
+
+  def resolve_and_copy(self, key):
+    # Resolve key to system package name
+    # TODO: Cross-platform solution for args
+    key_entry = bloom.generators.common.resolve_rosdep_key(key, "ubuntu", "trusty")
```

Comment:
Aren't we aiming for `vivid` rather then `trusty`? Couldn't this also be determined based on the system the script is run on?